Sources/App/Core/MarkdownHTMLConverter.swift

```diff
@@ -0,0 +1,80 @@
+import libcmark_gfm
+
+struct MarkdownHTMLConverter {
+    
+    enum ConversionError: Error {
+        case failedToCreateParser
+        case markdownToASTError
+        case astRenderError
+        case htmlToStringFailed
+    }
+    
+    let markdown: String
+    let enableGFM: Bool
+    
+    let options = CMARK_OPT_SAFE
+    
+    private func createCommonMarkNode(markdown: String) throws -> UnsafeMutablePointer<cmark_node> {
+        let tree = markdown.withCString { (document: UnsafePointer<Int8>) -> UnsafeMutablePointer<cmark_node>? in
+            let stringLength = Int(strlen(document))
+            return cmark_parse_document(document, stringLength, options)
+        }
+        
+        guard let ast = tree else {
+            throw ConversionError.markdownToASTError
+        }
+        
+        return ast
+    }
+    
+    private func createGFMCommonMarkNode(markdown: String) throws -> UnsafeMutablePointer<cmark_node> {
+        cmark_gfm_core_extensions_ensure_registered()
+        
+        guard let parser = cmark_parser_new(options) else {
+            throw ConversionError.failedToCreateParser
+        }
+        
+        defer {
+            cmark_parser_free(parser)
+        }
+        
+        ["table", "strikethrough", "tasklist", "tagfilter", "autolink"].forEach {
+            if let syntaxExtension = cmark_find_syntax_extension($0) {
+                cmark_parser_attach_syntax_extension(parser, syntaxExtension)
+            }
+        }
+        
+        let tree = markdown.withCString { (document: UnsafePointer<Int8>) -> UnsafeMutablePointer<cmark_node>? in
+            let stringLength = Int(strlen(document))
+            cmark_parser_feed(parser, document, stringLength)
+            return cmark_parser_finish(parser)
+        }
+        
+        guard let ast = tree else {
+            throw ConversionError.markdownToASTError
+        }
+        
+        return ast
+    }
+    
+    func toHTML() throws -> String {
+        let markdown = self.markdown.replaceShorthandEmojis()
```

Comment:
Reminder to check this but perhaps you know, @Sherlouk - does the `:emoji:` substitution perhaps come for free with this parser?

Reply:
I don't believe so - I noticed in the snapshot tests even after I updated to GFM that the emojis weren't being handled!